File: lang/script/go/fizzbuzz2.go
package main
import "fmt"
func main() {
	for i := 1; i <= 100; i++ {
		s := ""
		if 0 == (i % 3) {
			s = "Fizz"
		}
		if 0 == (i % 5) {
			s = s + "Buzz"
		}
		if "" == s {
			fmt.Print( i )
		} else {
			fmt.Print( s )
		}
		fmt.Print( " " )
	}
	fmt.Println( "" )
}

File: lang/script/ruby/fizzbuzz2.rb
#! /usr/bin/env ruby

for i in (1..100)
  s = ""
  if 0 == (i % 3) then
    s = "Fizz"
  end
  if 0 == (i % 5) then
    s = s + "Buzz"
  end

  if "" == s then
    print i
  else
    print s
  end
  print " "
end
print "\n"

